unity-client/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Common/AssetPromiseKeeper.cs

```diff
@@ -217,26 +246,48 @@ private void ForgetBlockedPromises(object loadedPromiseId)
                 var promise = blockedPromisesToForget[i];
                 promise.ForceFail();
                 Forget(promise);
+
+                var enumerator = SkipFrameIfOverBudget();
+
+                if (enumerator != null)
+                    yield return enumerator;
             }
         }
 
-        private void LoadBlockedPromises(object loadedPromiseId)
+        private List<AssetPromiseType> GetBlockedPromisesToLoadForId(object masterPromiseId)
         {
-            List<AssetPromiseType> blockedPromisesToLoad = new List<AssetPromiseType>();
+            var blockedPromisesToLoadAux = new List<AssetPromiseType>();
 
-            using (var iterator = masterToBlockedPromises[loadedPromiseId].GetEnumerator())
+            using (var iterator = masterToBlockedPromises[masterPromiseId].GetEnumerator())
             {
                 while (iterator.MoveNext())
                 {
                     var blockedPromise = iterator.Current;
 
-                    if (blockedPromise.state == AssetPromiseState.WAITING)
-                        blockedPromisesToLoad.Add(blockedPromise);
-
+                    blockedPromisesToLoadAux.Add(blockedPromise);
                     blockedPromises.Remove(blockedPromise);
                 }
             }
 
+            blockedPromisesToLoadAux.Sort(PromiseSortAlgorithm);
+
+            return blockedPromisesToLoadAux;
+        }
+
+        protected virtual int PromiseSortAlgorithm(AssetPromiseType promiseA, AssetPromiseType promiseB)
+        {
+            return 0;
+        }
```

Comment:
there is no comparison here, is it fine like this? it seems odd

Reply:
Oops, this is from the sort by distance PR. removing it.